### PR TITLE
Ftr/triple

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
- Ftr: http -> triple using dynamic API generation support 
- Transfer dubbo-go server's ip:port from API to Client. 
- Import formatted 
- Nacos registry support
- HTTP random loadbalancer 
- go version upgrate to 1.15

### CODE REVIEW

File: pkg/adapter/dubboregistry/registry/registry.go [hunk 1]
```diff
@@ -75,9 +77,18 @@ func GetRegistry(name string, regConfig model.Registry, listener common2.Registr
 }
 
 // CreateAPIConfig returns router.API struct base on the input
-func CreateAPIConfig(urlPattern string, dboBackendConfig config.DubboBackendConfig, methodString string, mappingParams []config.MappingParam) router.API {
+func CreateAPIConfig(urlPattern, location string, dboBackendConfig config.DubboBackendConfig, methodString string, mappingParams []config.MappingParam) router.API {
 	dboBackendConfig.Method = methodString
```
Comment (MasterKenway) [inline pkg/adapter/dubboregistry/registry/registry.go:81]: What is the purpose for the input param named `location`?

File: pkg/client/triple/triple.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	}
+	if err := p.Connect(context.Background(), targetURL); err != nil {
+		panic(err)
+	}
+	meta := make(map[string][]string)
+	reqData, _ := ioutil.ReadAll(req.IngressRequest.Body)
+	call, err := p.Call(context.Background(), req.API.Method.IntegrationRequest.Interface, req.API.Method.IntegrationRequest.Method, reqData, (*proxymeta.Metadata)(&meta))
+	if err != nil {
+		panic(err)
+	}
```
Comment (MasterKenway) [inline pkg/client/triple/triple.go]: panic here?

File: pkg/config/api_config.go [hunk 3]
```diff
@@ -27,8 +27,11 @@ import (
 
 import (
 	fc "github.com/dubbogo/dubbo-go-pixiu-filter/pkg/api/config"
+
 	etcdv3 "github.com/dubbogo/gost/database/kv/etcd/v3"
+
 	perrors "github.com/pkg/errors"
+
 	"go.etcd.io/etcd/api/v3/mvccpb"
 )
```
Comment (MasterKenway) [inline pkg/config/api_config.go:36]: Those blank line kind of strange, isn't it ?

File: pkg/server/api_config_manager.go [hunk 4]
```diff
@@ -22,13 +22,14 @@ import (
...
+	"github.com/dubbogo/dubbo-go-pixiu-filter/pkg/router"
+
 	"github.com/pkg/errors"
 )
 
 import (
 	"github.com/apache/dubbo-go-pixiu/pkg/model"
-	"github.com/dubbogo/dubbo-go-pixiu-filter/pkg/api/config"
-	"github.com/dubbogo/dubbo-go-pixiu-filter/pkg/router"
 )
```
Comment (MasterKenway) [inline pkg/server/api_config_manager.go:34]: Imports here is not correct as well

**Comment by MasterKenway** [comment]
@PhilYue As we discussed tonight, after server reflection is done, the `github.com/mercari/grpc-http-proxy` could be replaced

File: pkg/adapter/dubboregistry/registry/registry.go (hunk 1 above)
Comment (LaurenceLiZhixin) [reply pkg/adapter/dubboregistry/registry/registry.go:81]: 对于自动生成 api config 的场景，在adaptor层已经拿到了provider 的全量信息，完全可以拿出来provider的地址作为api config的一部分，在下面协议层调用时，无需再次服务发现，只做负载均衡即可。
负载均衡也可以在 adaptor 层和协议层之间，以filter 的形式来做。 @MasterKenway

**Comment by LaurenceLiZhixin** [comment]
我今天搞下nacos 支持，以及负载均衡，还有上述问题。import 块的问题都是按照 import-formatter 来规范的，这块是否还有问题？

**Review by MasterKenway — APPROVED** [review]

File: pkg/config/api_config.go (hunk 3 above)
Comment (mark4z) [reply pkg/config/api_config.go:36]: > Those blank line kind of strange, isn't it ?

+1

Comment (LaurenceLiZhixin) [reply pkg/config/api_config.go:36]: Cai boss said follow the code formatting tool

File: pkg/client/triple/triple.go (hunk 2 above)
Comment (LaurenceLiZhixin) [reply pkg/client/triple/triple.go]: fixed

**Comment by LaurenceLiZhixin** [comment]
@MasterKenway I guess this pr is able to be merged.

**Review by ztelur — APPROVED** [review]

**Review by mark4z — APPROVED** [review]

File: pkg/adapter/dubboregistry/registry/nacos/interface_listener.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+func (z *nacosIntfListener) updateServiceList(serviceList []string) error {
+	// todo lock all svc listener
+
+	allSvcListener := z.reg.GetAllSvcListener()
+	subscribedServiceKeysMap := make(map[string]bool)
+	for k := range allSvcListener {
+		subscribedServiceKeysMap[k] = true
+	}
+	serviceNeedUpdate := make([]*serviceInfo, 0)
```
Comment (AlexStocks) [inline pkg/adapter/dubboregistry/registry/nacos/interface_listener.go:159]: make([]*serviceInfo, 0) 后面加这个 0 毫无意义

File: pkg/client/triple/triple.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"github.com/apache/dubbo-go-pixiu/pkg/client"
+)
+
+// InitDefaultTripleClient init default dubbo client
+func InitDefaultTripleClient() {
+	tripleClient = NewTripleClient()
+}
+
+var tripleClient *Client
```
Comment (AlexStocks) [inline pkg/client/triple/triple.go]: 第一个 clientOnc 贼难看，写全称吧 clientOnce
第二，var block
var （
tripleClient *Client
clientOnce sync.Once
)

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**